samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -0,0 +1,219 @@
+		if (rc == 0) {
+			LOG_INF("NUS BLE advertising started");
+		} else {
+			LOG_ERR("Failed to start NUS BLE advertising: %d", rc);
+		}
+	};
+	mAdvertisingRequest.onStopped = []() { LOG_DBG("NUS BLE advertising stopped"); };
+
+	settings_load();
+#if defined(CONFIG_BT_FIXED_PASSKEY)
+	VerifyOrReturnError(bt_passkey_set(CONFIG_CHIP_NUS_FIXED_PASSKEY) == 0, false);
+#endif
+	VerifyOrReturnError(bt_conn_auth_cb_register(&sConnAuthCallbacks) == 0, false);
+	VerifyOrReturnError(bt_conn_auth_info_cb_register(&sConnAuthInfoCallbacks) == 0, false);
+	VerifyOrReturnError(bt_nus_init(&sNusCallbacks) == 0, false);
+
+	return true;
+}
+
+void NUSService::StartServer()
+{
+	if (mIsStarted) {
+		LOG_WRN("NUS service was already started");
+		return;
+	}
+
+	PlatformMgr().LockChipStack();
+	CHIP_ERROR ret = BLEAdvertisingArbiter::InsertRequest(mAdvertisingRequest);
+	PlatformMgr().UnlockChipStack();
+
+	if (CHIP_NO_ERROR != ret) {
+		LOG_ERR("Could not start NUS service");
+		mIsStarted = false;
+		return;
+	}
+	mIsStarted = true;
+	LOG_INF("NUS service started");
+}
+
+void NUSService::StopServer()
+{
+	VerifyOrReturn(mIsStarted);
+
+	PlatformMgr().LockChipStack();
+	BLEAdvertisingArbiter::CancelRequest(mAdvertisingRequest);
+	PlatformMgr().UnlockChipStack();
+
+	mIsStarted = false;
+	LOG_INF("NUS service stopped");
+}
+
+void NUSService::RxCallback(struct bt_conn *conn, const uint8_t *const data, uint16_t len)
+{
+	VerifyOrReturn(bt_conn_get_security(GetNUSService().mBTConnection) >= BT_SECURITY_L2);
+	LOG_DBG("NUS received: %d bytes", len);
+	GetNUSService().DispatchCommand(reinterpret_cast<const char *>(data), len);
+}
+
+bool NUSService::RegisterCommand(const char *const name, size_t length, CommandCallback callback, void *context)
+{
+	VerifyOrReturnError(name, false);
+	VerifyOrReturnError(mCommandsCounter <= CONFIG_CHIP_NUS_MAX_COMMANDS, false);
+
+	struct Command newCommand {};
+	memcpy(newCommand.command, name, length);
+	newCommand.commandLen = length;
+	newCommand.callback = callback;
+	newCommand.context = context;
+	mCommandsList[mCommandsCounter++] = newCommand;
+
+	return true;
+}
+
+void NUSService::DispatchCommand(const char *const data, uint16_t len)
+{
+	for (const auto &c : mCommandsList) {
+		if (strncmp(data, c.command, len) == 0 && len == strlen(c.command)) {
+			if (c.callback) {
+				PlatformMgr().LockChipStack();
+				c.callback(c.context);
+				PlatformMgr().UnlockChipStack();
+			}
+			return;
+		}
+	}
+	LOG_ERR("NUS command unkown!");
+}
+
+bool NUSService::SendData(const char *const data, size_t length)
+{
+	VerifyOrReturnError(mIsStarted && mBTConnection, false);
+	VerifyOrReturnError(mBTConnection, false);
+	VerifyOrReturnError(bt_conn_get_security(GetNUSService().mBTConnection) >= BT_SECURITY_L2, false);
+	VerifyOrReturnError(bt_nus_send(mBTConnection, reinterpret_cast<const uint8_t *const>(data), length) == 0,
+			    false);
+	return true;
+}
+
+void NUSService::Connected(struct bt_conn *conn, uint8_t err)
```

Comment:
```suggestion
void NUSService::Connected(bt_conn *conn, uint8_t err)
```
nit: all those `struct` are not C++-idiomatic.